### PR TITLE
Add some test

### PR DESCRIPTION
**Pull request by npes87184**
- test_strquote
- test_buffer_util

### CODE REVIEW

**Comment by rom1v** [comment]
:+1: Cool, more tests! :tada:

I will review this soon.

File: app/tests/test_strutil.c [hunk 1]
```diff
@@ -126,6 +127,19 @@ static void test_xstrjoin_truncated_after_sep(void) {
...
 
+static void test_strquote(void) {
+    const char *s = "abcde";
+    char *out = strquote(s);
+
+    // add '"' in the begin and the end, also check out is null terminated
+    assert((strlen(s) + 2) == strlen(out));
+
+    // check only add '"' in the begin and the end
+    assert(!strncmp(s, out + 1, strlen(s)));
```
Comment (rom1v) [inline app/tests/test_strutil.c]: I suggest to replace this by a single `strcmp()` call:

```diff
diff --git a/app/tests/test_strutil.c b/app/tests/test_strutil.c
index f4afabd..1489b1b 100644
--- a/app/tests/test_strutil.c
+++ b/app/tests/test_strutil.c
@@ -131,11 +131,8 @@ static void test_strquote(void) {
     const char *s = "abcde";
     char *out = strquote(s);
 
-    // add '"' in the begin and the end, also check out is null terminated
-    assert((strlen(s) + 2) == strlen(out));
-
-    // check only add '"' in the begin and the end
-    assert(!strncmp(s, out + 1, strlen(s)));
+    // add '"' at the beginning and the end
+    assert(!strcmp("\"abcde\"", out));
 
     SDL_free(out);
 }
```

It catches more possible errors (it asserts the expected output, not only some of its characteristics).

File: app/src/util/buffer_util.h [hunk 2]
```diff
@@ -36,8 +36,8 @@ buffer_read32be(const uint8_t *buf) {
     return (buf[0] << 24) | (buf[1] << 16) | (buf[2] << 8) | buf[3];
 }
 
-static inline
-uint64_t buffer_read64be(const uint8_t *buf) {
+static inline uint64_t
+buffer_read64be(const uint8_t *buf) {
```
Comment (rom1v) [inline app/src/util/buffer_util.h]: :+1:

File: app/tests/test_buffer_util.c [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+static void test_buffer_read64be(void) {
+    uint64_t val;
+    uint8_t buf[8] = {0xAB, 0xCD, 0x12, 0x34,
+                      0x56, 0x78, 0x90, 0xEF};
+
+    val = buffer_read64be(buf);
+
+    assert(val == 0xABCD1234567890EF);
+}
```
Comment (rom1v) [inline app/tests/test_buffer_util.c]: :+1:

(nitpick: I would just change the code style a bit)

```diff
diff --git a/app/tests/test_buffer_util.c b/app/tests/test_buffer_util.c
index 3d2cf3b..ba3f9f0 100644
--- a/app/tests/test_buffer_util.c
+++ b/app/tests/test_buffer_util.c
@@ -41,29 +41,26 @@ static void test_buffer_write64be(void) {
 }
 
 static void test_buffer_read16be(void) {
-    uint16_t val;
     uint8_t buf[2] = {0xAB, 0xCD};
 
-    val = buffer_read16be(buf);
+    uint16_t val = buffer_read16be(buf);
 
     assert(val == 0xABCD);
 }
 
 static void test_buffer_read32be(void) {
-    uint32_t val;
     uint8_t buf[4] = {0xAB, 0xCD, 0x12, 0x34};
 
-    val = buffer_read32be(buf);
+    uint32_t val = buffer_read32be(buf);
 
     assert(val == 0xABCD1234);
 }
 
 static void test_buffer_read64be(void) {
-    uint64_t val;
     uint8_t buf[8] = {0xAB, 0xCD, 0x12, 0x34,
                       0x56, 0x78, 0x90, 0xEF};
 
-    val = buffer_read64be(buf);
+    uint64_t val = buffer_read64be(buf);
 
     assert(val == 0xABCD1234567890EF);
 }
```

**Comment by npes87184** [comment]
LGTM!

**Comment by rom1v** [comment]
Merged on dev:
 - 5eeaed09aed64b088cbc7cf946923aeea92e4f67
 - b2bf25c52ca08d8de812186ab7651128e9989cb3
 - fbc86a616c38ea91beee586c417ea405ea6be56a

**Pull request closed without merging**